frontend/src/components/pages/connect/helper.tsx

```diff
@@ -608,14 +546,75 @@ export class ConfirmModal<T> extends Component<{
 
         // render error object
         return {
-            title: <div style={{ display: 'inline-flex', alignItems: 'center', gap: '0.3em' }}>
+            title: <div style={{display: 'inline-flex', alignItems: 'center', gap: '0.3em'}}>
                 <span>Error</span>
-                <span style={{ fontSize: '75%', opacity: 0.7 }}>- {apiErr.statusCode}</span>
+                <span style={{fontSize: '75%', opacity: 0.7}}>- {apiErr.statusCode}</span>
             </div>,
-            content: <div className="codeBox" style={{ mixBlendMode: 'multiply' }}>{apiErr.message}</div>
+            content: <div className="codeBox" style={{mixBlendMode: 'multiply'}}>{apiErr.message}</div>
         };
     }
-}
+
+    const cancel = action(() => {
+        runInAction(() => {
+            $state.isPending = false;
+            $state.error = null;
+            props.clearTarget();
+        })
+    })
+
+    const success = action((target: T) => {
+        const messageContent = props.successMessage(target);
+        toast({
+            status: 'success',
+            description: messageContent,
+        });
+
+        cancel();
+    })
+
+    const onOk = action(async () => {
+        $state.isPending = true;
+        const target = props.target()!;
+        try {
+            await props.onOk(target);
+            success(target);
+        } catch (err) {
+            $state.error = err as any;
+        } finally {
+            $state.isPending = false;
+        }
+    })
+
+    const target = props.target();
+    const err = renderError();
+
+    const content = target && props.content(target);
+
+    return <Modal
+        className="confirmModal"
+        open={target != null}
+        centered closable={false} maskClosable={!$state.isPending} keyboard={!$state.isPending}
+        okText={$state.error ? 'Retry' : 'Yes'}
+        confirmLoading={$state.isPending}
+        okType="danger"
+        cancelText="No"
+        cancelButtonProps={{ disabled: $state.isPending }}
+        onCancel={cancel}
+        onOk={onOk}
+    >
+        <>
+            <div>
```

Comment:
[suggestion] Use Box

Reply:
This is not related to a Toast, I'm sure it's addressed in #847